test/e2e/event_test.go

```diff
@@ -0,0 +1,148 @@
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/events"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/spf13/cobra"
+)
+
+func TestCli_Events(t *testing.T) {
+	subCommand := "event"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  events.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "get"})
+				rescueStdout := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				w.Close()
+				out, _ := io.ReadAll(r)
+				os.Stdout = rescueStdout
+				if !strings.Contains(string(out[:]), "instance.created") &&
+					!strings.Contains(string(out[:]), "instance.deleted") &&
+					!strings.Contains(string(out[:]), "ssh_key.created") &&
+					!strings.Contains(string(out[:]), "ssh_key.deleted") {
+					t.Error("expected output should include instance.created, instance.deleted, ssh_key.created, ssh_key.deleted by TYPE")
+				}
+			},
+		},
+		{
+			name: "get_by_orgID",
+			fields: fields{
+				MainCmd:  events.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "get", "-o", os.Getenv("METAL_ORG_ID")})
+				rescueStdout := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				w.Close()
+				out, _ := io.ReadAll(r)
+				os.Stdout = rescueStdout
+				if !strings.Contains(string(out[:]), "instance.created") &&
+					!strings.Contains(string(out[:]), "instance.deleted") &&
+					!strings.Contains(string(out[:]), "ssh_key.created") &&
+					!strings.Contains(string(out[:]), "ssh_key.deleted") {
+					t.Error("expected output should include instance.created, instance.deleted, ssh_key.created, ssh_key.deleted by TYPE")
+				}
+			},
+		},
+		{
+			name: "get_by_projID",
+			fields: fields{
+				MainCmd:  events.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "get", "-p", os.Getenv("METAL_PROJ_ID")})
```

Comment:
In general we should create test resources rather than reading IDs from the environment.  Exceptions to that guideline include organization ID and hardware reservation ID, because those are harder or impossible to create dynamically.

I think a good set of tests for events would:
(1) create a test project (with a standard naming convention so we can easily identify test projects created by metal-cli tests)
(2) create a device in that test project
(3) run various `events` subcommands and validate that the output matches expectations

However: for now, I think it's sufficient to just have the first `get` test and save the more complex test cases for a later PR.

Reply:
Updated as per review comments.